### PR TITLE
expose controller initializers

### PR DESCRIPTION
**Pull request by deads2k**
Expose the initializers and resource availability for consistent composition and named controller exposure by downstream consumers.

### CODE REVIEW

File: cmd/kube-controller-manager/app/controllermanager.go [hunk 1]
```diff
@@ -292,7 +292,7 @@ var ControllersDisabledByDefault = sets.NewString(
 	"tokencleaner",
 )
 
-func newControllerInitializers() map[string]InitFunc {
+func NewControllerInitializers() map[string]InitFunc {
```
Comment (derekwaynecarr) [inline cmd/kube-controller-manager/app/controllermanager.go]: i would document why this is public and note to not change it ;-)

File: cmd/kube-controller-manager/app/controllermanager.go [hunk 2]
```diff
@@ -319,7 +319,7 @@ func newControllerInitializers() map[string]InitFunc {
 
 // TODO: In general, any controller checking this needs to be dynamic so
 //  users don't have to restart their controller manager if they change the apiserver.
-func getAvailableResources(clientBuilder controller.ControllerClientBuilder) (map[schema.GroupVersionResource]bool, error) {
+func GetAvailableResources(clientBuilder controller.ControllerClientBuilder) (map[schema.GroupVersionResource]bool, error) {
```
Comment (derekwaynecarr) [inline cmd/kube-controller-manager/app/controllermanager.go]: same here.

**Review by derekwaynecarr — CHANGES_REQUESTED** [review]
i am more concerned about this changing in future re-factors without a comment that says not to do it.

**Comment by deads2k** [comment]
> i am more concerned about this changing in future re-factors without a comment that says not to do it.

done

**Pull request merged**